plugins/embed-optimizer/lazy-load.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Lazy load embeds
+ *
+ * When an embed block is lazy loaded, the script tag is replaced with a script tag that has the original attributes
+ */
+
+const lazyEmbedsScripts = document.querySelectorAll(
+	'script[type="application/vnd.embed-optimizer.javascript"]'
+);
+const lazyEmbedScriptsByParents = new Map();
+
+const lazyEmbedObserver = new IntersectionObserver(
+	( entries ) => {
+		for ( const entry of entries ) {
+			if ( entry.isIntersecting ) {
+				const lazyEmbedParent = entry.target;
+				const lazyEmbedScript =
+					/** @type {HTMLScriptElement} */ lazyEmbedScriptsByParents.get(
+						lazyEmbedParent
+					);
+				const embedScript =
+					/** @type {HTMLScriptElement} */ document.createElement(
```

Comment:
I added this type comment because PhpStorm was alerting me of this:

![Argument type Element is not assignable to parameter type Node | string](https://github.com/user-attachments/assets/f066cd76-6b60-424b-998b-b404fb1b7644)

I'm not sure why PhpStorm thinks that `document.createElement()` can return a `string`.